lib/core/sync_job_runner.rb

```diff
@@ -86,10 +93,13 @@ def do_sync!
         ElasticConnectorActions.complete_sync(@connector_settings.id, job_id, @status.dup)
```

Comment:
Remind me again why we can't have `rescue *UNEXPECTED_APP_EXITS` in this function, and then just

```
ElasticConnectorActions.complete_sync(@connector_settings.id, job_id, {"error": "Connector service quit unexpectedly.")
raise
```

as its body?

Reply:
I've tried that, but it doesn't seem to catch the interrupt in these spawned job sync threads, but catches it in the parent thread 🤔 . I'll read up some stuff and do some local experiments. Not 100% sure, how interrupts get handled in Ruby (probably also specific to the ruby implementation?)